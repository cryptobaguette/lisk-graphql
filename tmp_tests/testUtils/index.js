const lisk = require('lisk-elements').default;

exports.liskApiCLient = lisk.APIClient.createTestnetAPIClient();

exports.apiFormatAccount = account => ({
  address: account.address,
  balance: account.balance,
  unconfirmedBalance: account.unconfirmedBalance,
  publicKey: account.publicKey,
  secondPublicKey: account.secondPublicKey || null,
});
