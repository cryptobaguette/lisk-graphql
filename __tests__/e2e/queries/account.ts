import { graphqlClient, liskNetwork } from '../../testUtils';

const address =
  liskNetwork === 'testnet' ? '6238004142195673234L' : '6687808873757044786L';
const publicKey =
  liskNetwork === 'testnet'
    ? 'd85d2106558fa3946c9b67fa7a8984f1b8d6e58e9ca559055555f5aee6aa280c'
    : '3a0ca5c44a51bc2076a749a492dc335cd1f8aff7624ae6b8e595c20f727e4952';

describe('account', () => {
  describe('args', () => {
    describe('address', () => {
      it('should return null if account not found', async () => {
        const query = `
          query account {
            account(address: "test") {
              address
            }
          }
        `;
        const data = await graphqlClient.request<{ account: any }>(query);
        // TODO check no errors for all the responses
        expect(data.account).toBeNull();
      });

      it('should fetch account with address', async () => {
        const query = `
          query account {
            account(address: "${address}") {
              address
            }
          }
        `;
        const data = await graphqlClient.request<{ account: any }>(query);
        expect(data.account.address).toBe(address);
      });
    });

    describe('publicKey', () => {
      it('should throw an error if publicKey is malformed', async () => {
        const query = `
          query account {
            account(publicKey: "test") {
              address
            }
          }
        `;
        try {
          await graphqlClient.request<{ account: any }>(query);
          throw new Error();
        } catch (error) {
          expect(error.message).toMatch('Invalid public key');
        }
      });

      it('should return null if account not found', async () => {
        const query = `
          query account {
            account(publicKey: "d85d2106558fa3946c9b67fa7a8984f1b8d6e58e9ca559055555f5aee6aa2815") {
              address
            }
          }
        `;
        const data = await graphqlClient.request<{ account: any }>(query);
        expect(data.account).toBeNull();
      });

      it('should fetch account with publicKey', async () => {
        const query = `
          query account {
            account(publicKey: "${publicKey}") {
              address
              publicKey
            }
          }
        `;
        const data = await graphqlClient.request<{ account: any }>(query);
        expect(data.account.address).toBeTruthy();
        expect(data.account.publicKey).toBe(publicKey);
      });
    });
  });
});
