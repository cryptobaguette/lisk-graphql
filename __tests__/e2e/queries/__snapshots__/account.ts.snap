// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account args publicKey should throw an error if publicKey is malformated 1`] = `"Invalid public key, must be 64 characters long: {\\"response\\":{\\"errors\\":[{\\"message\\":\\"Invalid public key, must be 64 characters long\\",\\"locations\\":[{\\"line\\":3,\\"column\\":13}],\\"path\\":[\\"account\\"],\\"extensions\\":{\\"code\\":\\"INTERNAL_SERVER_ERROR\\"}}],\\"data\\":{\\"account\\":null},\\"status\\":200},\\"request\\":{\\"query\\":\\"\\\\n          query account {\\\\n            account(publicKey: \\\\\\"test\\\\\\") {\\\\n              address\\\\n            }\\\\n          }\\\\n        \\"}}"`;
